### PR TITLE
Better error handling/reporting when loading saved/recent connections

### LINKED ISSUE #18321: Connection Dialog broken
Please add "Area" label and repro steps before submitting --> does not seem like I could do this?!

---
Version: 1.94.2 (user setup)
Commit: 384ff7382de624fb94dbaf6da11977bba1ecd427
Date: 2024-10-09T16:08:44.566Z
Electron: 30.5.1
ElectronBuildId: 10262041
Chromium: 124.0.6367.243
Node.js: 20.16.0
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.22621

SQL Server (mssql): v1.25.0

Steps to Reproduce:

1. add a new connection
2. Connection Dialog is empty or incomplete
 an error message in VSCode is shown: `Cannot read properties of undefined (reading 'displayName')`

Image
Image

### PR DESCRIPTION
**Pull request by Benjin**
Fixes #18321

### CODE REVIEW

File: src/connectionconfig/connectionDialogWebviewController.ts [hunk 1]
```diff
@@ -241,6 +241,8 @@ export class ConnectionDialogWebviewController extends ReactWebviewPanelControll
     private async initializeConnectionForDialog(
         connection: IConnectionInfo,
     ): Promise<IConnectionDialogProfile> {
+        console.debug(`Initializing connection: ${connection}`);
```
Comment (Benjin) [inline src/connectionconfig/connectionDialogWebviewController.ts:244]: remove

File: src/connectionconfig/connectionDialogWebviewController.ts [hunk 2]
```diff
@@ -1399,14 +1421,28 @@ export class ConnectionDialogWebviewController extends ReactWebviewPanelControll
...
-                    this.initializeConnectionForDialog(conn),
-                ),
+                recentConnections.map((conn) => {
+                    try {
+                        return this.initializeConnectionForDialog(conn);
+                    } catch (ex) {
+                        console.error(
+                            "Recent connection: " + getErrorMessage(ex),
+                        );
+                        return Promise.resolve(undefined);
```
Comment (Benjin) [inline src/connectionconfig/connectionDialogWebviewController.ts:1431]: `undefined`s need filtering out down here.

**Review by aasimkhan30 — APPROVED** [review]
Looks good. I am guessing these errors are

File: src/connectionconfig/connectionDialogWebviewController.ts [hunk 3]
```diff
@@ -359,33 +359,47 @@ export class ConnectionDialogWebviewController extends ReactWebviewPanelControll
...
+            const accounts =
+                await this._mainController.azureAccountService.getAccounts();
+            return accounts.map((account) => {
+                return {
+                    displayName: account.displayInfo.displayName,
+                    value: account.displayInfo.userId,
+                };
+            });
+        } catch (error) {
+            console.error(
```
[the remark this answers is not in the document]
Comment (Benjin) [reply src/connectionconfig/connectionDialogWebviewController.ts:372]: Judging from the types of error messages users reported in https://github.com/microsoft/vscode-mssql/issues/18321, I don't think these errors would be particularly useful to display to users: mostly "cannot read property 'someProp' of undefined" stemming from some stale stuff in the Entra cache.

This PR mostly makes sure that the entire connection dialog can take those failures (loading connections, querying Entra tenants) in stride without failing completely.

[the remark this answers is not in the document]
Comment (aasimkhan30) [reply src/connectionconfig/connectionDialogWebviewController.ts:372]: Makes sense. Last question, Does this mean that users won't be able to use the UI after we are done loading it? 
How do users recover from these issues?

**Review by lewis-sanchez — APPROVED** [review]

**Pull request merged**